libraries/chain/worker_evaluators.cpp

```diff
@@ -322,22 +322,22 @@ namespace golos { namespace chain {
         });
     }
 
-    void worker_result_approve_evaluator::do_apply(const worker_result_approve_operation& o) {
-        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_21__1013, "worker_result_approve_operation");
+    void worker_payment_approve_evaluator::do_apply(const worker_payment_approve_operation& o) {
+        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_21__1013, "worker_payment_approve_operation");
 
         auto approver_witness = _db.get_witness(o.approver);
         GOLOS_CHECK_LOGIC(approver_witness.schedule == witness_object::top19,
-            logic_exception::approver_of_result_should_be_in_top19_of_witnesses,
-            "Approver of result should be in Top 19 of witnesses");
+            logic_exception::approver_of_payment_should_be_in_top19_of_witnesses,
+            "Approver of payment should be in Top 19 of witnesses");
 
-        const auto& worker_result_post = _db.get_comment(o.author, o.permlink);
-        const auto& wto = _db.get_worker_result(worker_result_post.id);
+        const auto& wto_post = _db.get_comment(o.worker_techspec_author, o.worker_techspec_permlink);
+        const auto& wto = _db.get_worker_techspec(wto_post.id);
 
         const auto& wpo = _db.get_worker_proposal(wto.worker_proposal_post);
 
         GOLOS_CHECK_LOGIC(wto.state == worker_techspec_state::complete || wto.state == worker_techspec_state::payment,
-            logic_exception::worker_techspec_should_be_complete_or_paying_to_approve_result,
-            "Worker techspec should be complete or paying to approve result");
+            logic_exception::worker_techspec_should_be_complete_or_paying,
```

Comment:
`worker_techspec_state::work` ?
`worker_techspec_state::wip` ?

Reply:
@afalaleev In another pull request.